File: AoC2015/JavaScript/day1-2.js
//puzzle setup
const fs = require('fs');
var input = fs.readFileSync('./day1in.txt','utf8');
input = input.split('');

//do stuff
sum = 0;

for (i in input)
{
    input[i] == '(' ? sum++ : sum--;
    if (sum === -1)
    {
        console.log(Number(i)+1);
        break;
    }
}

File: AoC2015/Python3/day1-2.py
#puzzle setup
with open('./day1in.txt','r') as pInput:
    pInput = pInput.read()

#do stuff
mysum = 0
for i, c in enumerate(pInput, 1):
    mysum = (mysum + 1 if c == '(' else mysum - 1)
    if (mysum == -1):
        print(i)
        quit()
